lib/ansible/modules/web_infrastructure/ansible_tower/tower_organization.py

```diff
@@ -0,0 +1,141 @@
+    description:
+      description:
+        - String to use for the organization.
+      required: False
+      default: null
+    state:
+      description:
+        - Desired state of the resource.
+      required: False
+      default: "present"
+      choices: ["present", "absent"]
+    config_file:
+      description:
+        - Path to the Tower config file. See notes.
+      required: False
+      default: null
+
+
+requirements:
+  - "python >= 2.6"
+  - "ansible-tower-cli >= 3.0.3"
+
+notes:
+  - If no I(config_file) is provided we will attempt to use the tower-cli library
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format:
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+  - tasks
+    - name: Create organization
+      tower_organization:
+          name: "Foo"
+          description: "Foo bar organization"
+          state: present
+          config_file: "~/tower_cli.cfg"
+'''
+
+import os
+
+try:
+    import tower_cli
+    import tower_cli.utils.exceptions as exc
+    from tower_cli.utils import parser
+    from tower_cli.conf import settings
+
+    HAS_TOWER_CLI = True
+except ImportError:
+    HAS_TOWER_CLI = False
+
+
+def tower_auth_config(module):
+    config_file = module.params.get('config_file')
+    if not config_file:
+        return {}
+
+    config_file = os.path.expanduser(config_file)
+    if not os.path.exists(config_file):
+        module.fail_json(msg='file not found: %s' % config_file)
+    if os.path.isdir(config_file):
+        module.fail_json(msg='directory can not be used as config file: %s' % config_file)
+
+    with open(config_file, 'rb') as f:
+        return parser.string_to_dict(f.read())
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(),
+            description = dict(),
+            config_file = dict(),
+            state = dict(choices=['present', 'absent'], default='present'),
+        ),
+        supports_check_mode=False
+    )
+
+    if not HAS_TOWER_CLI:
+        module.fail_json(msg='ansible-tower-cli required for this module')
+
+    name = module.params.get('name')
+    description = module.params.get('description')
+    state = module.params.get('state')
+
+    json_output = {'organization': name, 'state': state}
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        organization = tower_cli.get_resource('organization')
+        try:
+            if state == 'present':
+                result = organization.modify(name=name, description=description, create_on_missing=True)
+                json_output['id'] = result['id']
+            elif state == 'absent':
+                result = organization.delete(name=name)
+        except (exc.ConnectionError, exc.BadRequest) as excinfo:
+            module.fail_json(msg='{}'.format(excinfo), changed=False)
```

Comment:
a better message might start 'Failed to update the organization: <execinfo>' ... also It would be nice to make the distinction between retrieval and update, as they could pose separate permission issues. i.e. I can read but not update the organization.